### ISSUE TITLE
tempDouble undefined

### ISSUE THREAD

**Issue by d3x0r** [emscripten-core/emscripten#8774]
```
emcc --version
emcc (Emscripten gcc/clang-like replacement) 1.38.33 (commit 963d839afa49c7496cfb3fe5427c11e29e190130)
```

I'm compiling with -g and no -O flags... and loading the .js/.wasm with import.

Stat uses a variable 'tempDouble', and `tempI64` which don't exist. 
I'm guessing it didn't show up as an error before because not 'use strict' which loading as a module implies?

```
doStat:function (func, path, buf) {
                (tempI64 = [stat.size>>>0,(tempDouble=stat.size,(+(Math_abs(tempDouble))) >= 1.0 ? (tempDouble > 0.0 ? ((Math_min((+(Math_floor((tempDouble)/4294967296.0))), 4294967295.0))|0)>>>0 : (~~((+(Math_ceil((tempDouble - +(((~~(tempDouble)))>>>0))/4294967296.0)))))>>>0) : 0)],HEAP32[(((buf)+(40))>>2)]=tempI64[0],HEAP32[(((buf)+(44))>>2)]=tempI64[1]);

}
```

**Comment by kripken** [comment]
Thanks, this does look like a bug. I have a wip PR, but I'm not sure how to test it - when I add 'use strict' it still doesn't error on those two not being defined. How exactly did you get the error to happen (what does "with import" mean)?

**Comment by d3x0r** [comment]
It didn't actually show up until I used the stat function... 
This is the stack dump that comes with it...

```
sack.vfs.wasm.js:9199 ReferenceError: tempDouble is not defined
    at Object.doStat (sack.vfs.wasm.js:4786)
    at ___syscall195 (sack.vfs.wasm.js:6209)
    at _stat (:8087/wasm-function[3070]:64)
    at _IsPath (:8087/wasm-function[334]:61)
    at __fs_ExpandVolume (:8087/wasm-function[495]:598)
    at _sack_vfs_fs_load_crypt_volume (:8087/wasm-function[501]:221)
    at _Volume (:8087/wasm-function[2957]:442)
    at Object.Module._Volume (sack.vfs.wasm.js:8526)
    at new Volume (sack.vfs.wasm.js:1703)
    at Object.Volume (sack.vfs.wasm.js:1703)
```


I have post the sources, and compiled project at 
https://d3x0r.org:444/javascript/wasmfs/
I didn't include an index.html so that directory is browsable.... I included `test-wasm-fs.zip` which is all the files in that (except itself).

emk.bat is the build script   (the version in the .zip has more remarked-out lines... I cleaned it up a little in the /wasmfs/ folder.

Index.html has...

```
<SCRIPT type="module" src="test.js"> </SCRIPT>
```
test.js has...
```
	import { Module as vfsModule } from "./sack.vfs.wasm.js" 
```

I added a 'var tempDouble, tempI64;' to my `exportTail.js`  ... it's commented out now, so it can rebuild with the missing symbols.

I'm sorry this isn't a more concise demonstration...  but it is just compiling 3 sources together :)

**Comment by kripken** [comment]
Thanks @d3x0r, now I see the error. Fix + test in #8965
